pkg/controllers/execution/execution_controller.go

```diff
@@ -271,11 +271,7 @@ func (c *Controller) updateAppliedCondition(work *workv1alpha1.Work, status meta
 		if updateErr == nil {
 			return nil
 		}
-		updated := &workv1alpha1.Work{}
-		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: work.Namespace, Name: work.Name}, updated); err == nil {
-			// make a copy, so we don't mutate the shared cache
-			work = updated.DeepCopy()
-		} else {
+		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: work.Namespace, Name: work.Name}, work); err != nil {
```

Comment:
Here will mutate the memory pointed by the `work` pointer, which might be dangerous as we don't know how the `work` be used after `updateAppliedCondition`.
```suggestion
		work = &workv1alpha1.Work{}
		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: work.Namespace, Name: work.Name}, work); err != nil {
			klog.Errorf("Failed to get updated work %s/%s: %v", work.Namespace, work.Name, err)
		}
```
How about initializing a block of memory specifically for this function to use?